src/init/AppRunner.ts

```diff
@@ -158,20 +184,49 @@ export class AppRunner {
   }
 
   /**
-   * Handles the first Components.js instantiation,
-   * where CLI settings and variable mappings are created.
+   * Handles the first Components.js instantiation.
+   * Uses it to extract the CLI settings and use those to create variable bindings.
    */
-  private async resolveVariables(componentsManager: ComponentsManager<CliResolver>, argv: string[]):
+  private async cliToVariables(componentsManager: ComponentsManager<CliResolver>, argv: string[]):
   Promise<VariableBindings> {
+    const cliResolver = await this.createCliResolver(componentsManager);
+    const settings = await this.extractCliSettings(cliResolver.cliExtractor, argv);
+    return await this.resolveSettings(cliResolver.settingsResolver, settings);
```

Comment:
I have seen the above code already somewhere else; duplication?

Reply:
There is a small difference: the one here creates a CLI resolver, extracts the CLI settings, and then converts those settings to variable bindings. The other one creates a CLI resolver, and immediately applies provided settings (and then merges them with the provided variable bindings).

Reply:
Maybe just add that as a comment above the block

```
//  Creates CLI resolver and immediately apply provided [settings|shorthand parameters]
```